src/carriers/zfp_portmonitor/CMakeLists.txt

```diff
@@ -8,24 +8,34 @@ yarp_prepare_plugin(zfp TYPE ZfpMonitorObject
                         DEPENDS "CREATE_OPTIONAL_CARRIERS;ENABLE_yarpcar_portmonitor")
 
 if(NOT SKIP_zfp)
-  set(CMAKE_INCLUDE_CURRENT_DIR ON)
-  get_property(YARP_OS_INCLUDE_DIRS TARGET YARP_OS PROPERTY INCLUDE_DIRS)
-  get_property(YARP_sig_INCLUDE_DIRS TARGET YARP_sig PROPERTY INCLUDE_DIRS)
-  include_directories(${YARP_OS_INCLUDE_DIRS} ${YARP_sig_INCLUDE_DIRS})
-  include_directories(SYSTEM ${ZFP_INCLUDE_DIRS})
-
-  yarp_add_plugin(yarp_pm_zfp zfpPortmonitor.cpp
-                              zfpPortmonitor.h)
-  target_link_libraries(yarp_pm_zfp YARP_OS YARP_sig ${ZFP_LIBRARIES})
-
-  yarp_install(TARGETS yarp_pm_zfp
-               EXPORT YARP
-               COMPONENT runtime
-               LIBRARY DESTINATION ${YARP_DYNAMIC_PLUGINS_INSTALL_DIR}
-               ARCHIVE DESTINATION ${YARP_STATIC_PLUGINS_INSTALL_DIR})
-  yarp_install(FILES zfp.ini
-               COMPONENT runtime
-               DESTINATION ${YARP_PLUGIN_MANIFESTS_INSTALL_DIR})
-
-  set_property(TARGET yarp_pm_zfp PROPERTY FOLDER "Plugins/Port Monitor")
+
+    if(ZFP_FOUND)
```

Comment:
I don't like this option because if the `zfp` library is not found on the system, then the option for creating this plugin is not created and the plugin is not visible among the cmake options.
Therefore people does not even know that such a plugin exists, and there is no information about what to do to enable it.

Vice versa, the way I did the flag is visible, so people do know the plugin is available. If someone tries to use it but the library is not found, the error message will tell what to do.

This is far more user friendly.